pkg/storaged/utils.js

```diff
@@ -767,10 +925,12 @@ export function teardown_active_usage(client, usage) {
     // physical volumes here, and it is easiest to catch this
     // condition upfront by reshuffling the data structures.
 
-    function unmount(mounteds) {
-        return Promise.all(mounteds.map(m => {
-            return client.unmount_at(m.location, m.users);
-        }));
+    async function unmount(mounteds) {
+        for (const m of mounteds) {
+            await client.unmount_at(m.location, m.users);
+            if (m.set_noauto)
+                await set_fsys_noauto(client, m.block, m.location);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19388-20230928-071616-0731117d-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)